src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs

```diff
@@ -2205,5 +2220,37 @@ private static List<CompletionResult> CompleteFileNameAsCommand(CompletionContex
 
             return result;
         }
+
+        /// <summary>
+        /// Complete loop labels after labeled control flow statements such as Break and Continue.
+        /// </summary>
+        private static List<CompletionResult> CompleteLoopLabel(CompletionContext completionContext)
+        {
+            var result = new List<CompletionResult>();
+            foreach (Ast ast in completionContext.RelatedAsts)
+            {
+                if (ast is LabeledStatementAst labeledStatement
+                    && labeledStatement.Label is not null
+                    && (completionContext.WordToComplete is null || labeledStatement.Label.StartsWith(completionContext.WordToComplete, StringComparison.OrdinalIgnoreCase)))
+                {
+                    result.Add(new CompletionResult(labeledStatement.Label, labeledStatement.Label, CompletionResultType.Text, labeledStatement.Extent.Text));
+                }
+                else if (ast is ErrorStatementAst errorStatement)
+                {
+                    var labelMatch = Regex.Match(errorStatement.Extent.Text, @"(?<=^:)\w+(?=\s+(do|switch)\b(?!-))", RegexOptions.IgnoreCase);
```

Comment:
The regular expression seems to only handle `do{} while` and `switch { }` as special cases, and no other loop types, right?

I agree with @MartinGC94's [comment](https://github.com/PowerShell/PowerShell/pull/16438#issuecomment-1006010213) that a user would be most likely in an editor when working with loops. Handling the error statement like this is:
1. incomplete and not reliable using regex
2. add burden to maintenance (again, the regex)

So, I personally don't think we need to handle `ErrorStatementAst`.

Reply:
It only handles do/switch loops because those are the only loop types with this problem. Other loop types like: `:DemoLoop foreach ($x in $y) {break ` worked fine without this "hack" in my testing.

I don't mind removing it, but I disagree with point 1. From my testing it seems complete and reliable. You need to be in a context where break/continue are valid keywords and the errorstatement has to start at the loop label and look just like a real do/switch loop. Can you write silly stuff like `using :bla do{ break ` and still get completion? Sure, but I don't think that's a bad user experience because the intent is undeniably to write a loop and deleting 1 word or moving the loop down 1 line is enough to make it valid.

Reply:
@MartinGC94 Thanks for the clarification!
> It only handles do/switch loops because those are the only loop types with this problem. Other loop types like: `:DemoLoop foreach ($x in $y) {break` worked fine without this "hack" in my testing.

With this explained, I'm fine we keep the change. Can you please do the following:
1. Add a comment at the regex, to explain what is intended for this regex to catch. You already mentioned this in your comment, so capture it in the comment would be very helpful.
2. Add a few tests covering cases like `:DemoLoop foreach ($x in $y) {break` for other loops.